vocabulary-care-pathways/api/src/main/java/org/phenotips/vocabulary/internal/solr/CarePathwaysOntology.java

```diff
@@ -0,0 +1,496 @@
+import org.apache.solr.client.solrj.util.ClientUtils;
+import org.apache.solr.common.SolrInputDocument;
+import org.apache.solr.common.params.DisMaxParams;
+import org.apache.solr.common.params.SpellingParams;
+
+/**
+ * Provides access to the Care Pathways care and tests vocabulary. The vocabulary prefix is {@code CP}.
+ *
+ * @version $Id$
+ * @since 1.0
+ */
+@Component
+@Named("care-pathways")
+@Singleton
+public class CarePathwaysOntology extends AbstractCSVSolrVocabulary
+{
+    /** The category for care pathways test-related terms. */
+    private static final String TEST_CATEGORY = "test";
+
+    /** The category for care pathways care-related terms. */
+    private static final String CARE_CATEGORY = "care";
+
+    /** The label for the field storing ancestors for term. */
+    private static final String TERM_CATEGORY = "term_category";
+
+    /** The standard prefix for the vocabulary terms. */
+    private static final String STANDARD_PREFIX = "CP";
+
+    /** A colon. */
+    private static final String COLON = ":";
+
+    /** The filter for obtaining only the test-related terms. */
+    private static final String DEFAULT_TEST_FILTER = TERM_CATEGORY + COLON + STANDARD_PREFIX + "\\:1";
+
+    /** The filter for obtaining only the care-related terms. */
+    private static final String DEFAULT_CARE_FILTER = TERM_CATEGORY + COLON + STANDARD_PREFIX + "\\:2";
+
+    /** The label for the field storing ID for term. */
+    private static final String ID = "id";
+
+    /** The label for the field storing term name. */
+    private static final String NAME = "name";
+
+    /** The label for the field storing parents for term. */
+    private static final String IS_A = "is_a";
+
+    /** The open character. */
+    private static final String OPEN = "[";
+
+    /** The close character. */
+    private static final String CLOSE = "]";
+
+    /** The list of supported categories for this vocabulary. */
+    private static final Collection<String> SUPPORTED_CATEGORIES = Arrays.asList(TEST_CATEGORY, CARE_CATEGORY);
+
+    /** For determining if a query is a an id. */
+    private static final Pattern ID_PATTERN = Pattern.compile("^CP:[0-9]+$", Pattern.CASE_INSENSITIVE);
+
+    @Override
+    public String getIdentifier()
+    {
+        return "care-pathways";
+    }
+
+    @Override
+    public String getName()
+    {
+        return "Care pathways tests and patient care";
+    }
+
+    @Override
+    public Set<String> getAliases()
+    {
+        Set<String> aliases = new HashSet<>();
+        aliases.add(getIdentifier());
+        aliases.add(STANDARD_PREFIX);
+        return aliases;
+    }
+
+    @Override
+    public Collection<String> getSupportedCategories()
+    {
+        return SUPPORTED_CATEGORIES;
+    }
+
+    @Override
+    public String getDefaultSourceLocation()
+    {
+        // Is empty, as there is no default location for the vocabulary source file.
+        return StringUtils.EMPTY;
```

Comment:
The resources are not part of the installation in any way, which makes this not trivial to install. Can this be done instead:

- also include the current TSV source in a jar, either this one or as a dependency to the resources jar
- return something like  `getClass().getClassLoader().getSystemClassLoader().getResource("CarePathwaysOrderedTestsAndCare.tsv").toString()` here
- a replacement for the bundled version can either be placed in `WEB-INF/classes`, or anywhere else and the correct path would be entered in the reindex page